src/main/java/com/spotify/futures/CompletableFuturesExtra.java

```diff
@@ -46,6 +47,23 @@ private CompletableFuturesExtra() {
     return new ListenableToCompletableFutureWrapper<V>(future);
   }
 
+  /**
+   * Wrap a {@link ListenableFuture} in a {@link CompletableFuture}. The returned future will
+   * complete with the same result or failure as the original future. Completing the returned
+   * future does not complete the original future.
+   *
+   * @param future The {@link ListenableFuture} to wrap in a {@link CompletableFuture}.
+   * @param executor The executor to run the wrapped {@code future} in.
+   * @return A {@link CompletableFuture} that completes when the original future completes.
+   */
+  public static <V> CompletableFuture<V> toCompletableFuture(
+      ListenableFuture<V> future, Executor executor) {
+    if (future instanceof CompletableToListenableFutureWrapper) {
+      return ((CompletableToListenableFutureWrapper<V>) future).unwrap();
```

Comment:
This code path means that it may not always run on the executor, which can be confusing to users.